pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -0,0 +1,166 @@
+//go:build linux
+// +build linux
+
+package cadvisor //nolint:golint
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/google/cadvisor/cache/memory"
+	"github.com/google/cadvisor/container"
+	v2 "github.com/google/cadvisor/info/v2"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/metrics"
+	"github.com/google/cadvisor/storage"
+	"github.com/google/cadvisor/utils/sysfs"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+
+	"github.com/grafana/agent/pkg/integrations"
+
+	// Register container providers
+
+	"github.com/google/cadvisor/container/containerd"
+	_ "github.com/google/cadvisor/container/containerd/install" // register containerd container plugin
+	_ "github.com/google/cadvisor/container/crio/install"       // register crio container plugin
+	"github.com/google/cadvisor/container/docker"
+	_ "github.com/google/cadvisor/container/docker/install" // register docker container plugin
+	"github.com/google/cadvisor/container/raw"
+	_ "github.com/google/cadvisor/container/systemd/install" // register systemd container plugin
+)
+
+// Matching the default disabled set from cadvisor - https://github.com/google/cadvisor/blob/3c6e3093c5ca65c57368845ddaea2b4ca6bc0da8/cmd/cadvisor.go#L78-L93
+var disabledMetricsSet container.MetricSet = container.MetricSet{
```

Comment:
Hmm, there's another way you can do this:

```go 
var disabledMetricsSet = *flag.Lookup("disable_metrics").Value.(*container.MetricsSet)
```

That would help you stay in sync with upstream.

Reply:
Ohh.. I see why I didn't do this. Because I'd have to import the `github.com/google/cadvisor/cmd` package. IIRC that caused some conflicts and wasn't worth the drama :(

Reply:
Ahhh, I didn't realize it was from another package. That's fine, this is good as-is.

Reply:
Put it back for now. Addressed everything else. When the build steps pass, I'm gonna go ahead and merge this.

Thanks for the repeated reviews @rfratto :)